mlir/examples/toy/Ch7/include/toy/Dialect.h

```diff
@@ -72,6 +72,9 @@ class StructType : public mlir::Type::TypeBase<StructType, mlir::Type,
 
   /// Returns the number of element type held by this struct.
   size_t getNumElementTypes() { return getElementTypes().size(); }
+
+  /// Returns the name of this struct type.
+  static constexpr StringRef getTypeName() { return "toy.struct"; }
```

Comment:
Why not `StringLiteral`?

Reply:
Oh yes, that makes more sense, I'll fix that!